03_letters/willshen8/letters.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func letters(s string) map[rune]int {
+	letterMap := make(map[rune]int)
+
+	for _, value := range s {
+		if letterFrequency, found := letterMap[value]; found {
```

Comment:
Take advantage of the default behavior of a `map` in go to return the "nil" value (in the case of int, it's 0) to avoid the need for branching.

Reply:
Thanks for this. I found an even easier way. just do `letterMap[value]++` and go increment it for you and no need to check for nil or 0.